### PR TITLE
docs: strimzi drainer cleaner

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

- Adds a new assembly to the _Managing Strimzi_ section in the _Using Strimzi_ guide.
- Add procedures to deploy and use Strimzi Drain Cleaner

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: documentation/assemblies/managing/assembly-drain-cleaner.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+The resource provides the configuration for registering the webhook with the Kubernetes API.
+
+The configuration defines the `rules` for the Kubernetes API to follow in the event of a pod eviction request.
+The rules specify that only `CREATE` operations related to `pods/eviction` sub-resources are intercepted.
+If these rules are met, the API forwards the notification.
+
+The `clientConfig` points to the Strimzi Drain Cleaner service and `/drainer` endpoint that exposes the webhook.
+The webhook uses a secure TLS connection, which requires authentication.
+The `caBundle` property specifies the certificate chain to validate HTTPS communication.
+Certificates are encoded in base64.
```
Comment (joaedwar) [inline documentation/assemblies/managing/assembly-drain-cleaner.adoc:35]:
```suggestion
Certificates are encoded in Base64.
```

File: documentation/assemblies/managing/assembly-drain-cleaner.adoc [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// This assembly is included in the following assemblies:
+//
+// using/assembly-management-tasks.adoc
+
+[id='assembly-drain-cleaner-{context}']
+= Evicting pods with Strimzi Drain Cleaner
+
+[role="_abstract"]
```
Comment (joaedwar) [inline documentation/assemblies/managing/assembly-drain-cleaner.adoc:8]: I think the abstract content in this assembly can actually be a module. There is a lot of conceptual info, in this I think.

File: documentation/modules/managing/proc-drain-cleaner-deploying.adoc [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-drain-cleaner.adoc
+
+[id='proc-drain-cleaner-deploying-{context}']
+= Deploying the Strimzi Drain Cleaner
+
+[role="_abstract"]
+Deploy the Strimzi Drain Cleaner to the Kubernetes cluster where the Cluster Operator and Kafka cluster is running.
```
Comment (joaedwar) [inline documentation/modules/managing/proc-drain-cleaner-deploying.adoc:9]:
```suggestion
Deploy the Strimzi Drain Cleaner to the Kubernetes cluster where the Cluster Operator and Kafka cluster are running.
```

Comment (joaedwar) [inline documentation/modules/managing/proc-drain-cleaner-deploying.adoc:9]:
```suggestion
Deploy the Strimzi Drain Cleaner to the Kubernetes cluster where the Cluster Operator and Kafka cluster are running.
```

File: documentation/modules/managing/proc-drain-cleaner-deploying.adoc [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            - "-Dquarkus.http.host=0.0.0.0"
+            - "--kafka"
+            - "--zookeeper" <1>
+          # ...
+----
+<1> Remove this option to exclude ZooKeeper from Strimzi Drain Cleaner operations.
+
+.Procedure
+
+. Configure a pod disruption budget of 0 (zero) for your Kafka deployment using `template` settings in the `Kafka` resource.
```
Comment (joaedwar) [inline documentation/modules/managing/proc-drain-cleaner-deploying.adoc:64]: I think the value should be in monospace `0`

File: documentation/modules/managing/proc-drain-cleaner-deploying.adoc [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+  # ...
+  zookeeper:
+    template:
+      podDisruptionBudget:
+        maxUnavailable: 0
+  # ...
+----
++
+Reducing the maximum pod disruption budget to zero prevents Kubernetes from automatically evicting (deleting) the pods in case of voluntary disruptions,
```
Comment (joaedwar) [inline documentation/modules/managing/proc-drain-cleaner-deploying.adoc:88]: Should choose either evicting or deleting . Parentheses are not needed.

File: documentation/modules/managing/proc-drain-cleaner-deploying.adoc [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+* If you are not using `cert-manager` with Kubernetes, apply the resources in the `/install/drain-cleaner/kubernetes` directory.
++
+[source,shell,subs="attributes+"]
+----
+kubectl apply -f ./install/drain-cleaner/kubernetes
+----
++
+The resources are configured with TLS certificates that have already been generated.
+Use these certificates if you are not changing any configuration for the Strimzi Drain Cleaner, such as namespaces, or service and pod names.
```
Comment (joaedwar) [inline documentation/modules/managing/proc-drain-cleaner-deploying.adoc:127]:
```suggestion
Use these certificates if you are not changing any configuration for the Strimzi Drain Cleaner, such as namespaces, services, and pod names.
```

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc [hunk 7]
```diff
@@ -8,6 +8,34 @@
 This procedure describes how to manually trigger a rolling update of an existing Kafka cluster or ZooKeeper cluster using a Kubernetes `Pod` annotation.
 When multiple pods from the same `StatefulSet` are annotated, consecutive rolling updates are performed within the same reconciliation run.
 
+.Prerequisites
+
+You can perform a rolling update on a Kafka cluster regardless of the topic replication factor used.
+But for Kafka to stay operational during the update, you'll need the following:
+
+* A highly available Kafka cluster deployment running with nodes that you wish to update.
```
Comment (joaedwar) [inline documentation/modules/managing/proc-manual-rolling-update-pods.adoc:16]: I think "wish" tends towards idiomatic. I think "would like to" might be a possible reword.

File: documentation/assemblies/managing/assembly-drain-cleaner.adoc [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+If your Kafka broker and ZooKeeper pods were deployed by Strimzi, you can use the Strimzi Drain Cleaner tool to handle the pod evictions.
+You need to set the `podDisruptionBudget` for your Kafka deployment to 0 (zero) for the Strimzi Drain Cleaner to work.
+
+By deploying the Strimzi Drain Cleaner, you can use the Cluster Operator to move Kafka pods instead of Kubernetes.
+Why is this beneficial? The Cluster Operator ensures that topics are never under-replicated.
+As Kubernetes worker nodes are drained consecutively, the Cluster Operator waits until topics are synchronized.
+This means that Kafka can remain operational during the eviction process.
+
+An admission webhook notifies the Strimzi Drain Cleaner of pod eviction requests to the Kubernetes API.
+The Strimzi Drain Cleaner then annotates the pods to be drained with a rolling update pod annotation.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-drain-cleaner.adoc:19]: I've updated to:

> The Strimzi Drain Cleaner then adds a rolling update annotation to the pods to be drained.

File: documentation/assemblies/managing/assembly-drain-cleaner.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-drain-cleaner.adoc:8]: Going to leave as is for now.

File: documentation/modules/managing/proc-drain-cleaner-deploying.adoc (hunk 6 above)
Comment (PaulRMellor) [reply documentation/modules/managing/proc-drain-cleaner-deploying.adoc:127]: The _name_ is important here, so changed to: _such as namespaces, service names, or pod names._

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc (hunk 7 above)
Comment (PaulRMellor) [reply documentation/modules/managing/proc-manual-rolling-update-pods.adoc:16]: _would like_ sounds good to me

**Comment by joaedwar** [comment]
LGTM

**Comment by PaulRMellor** [comment]
Hi @scholzj 
Unless we need another reviewer, we're ready to merge.
Please cherry-pick to 0.26 :+1:

**Pull request merged**